File: backend/src/main/java/com/castify/backend/models/playlist/PlaylistModel.java
package com.castify.backend.models.playlist;

import com.castify.backend.entity.PlaylistItem;
import com.castify.backend.models.user.UserModel;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class PlaylistModel {
    private String id;
    private String name;
    private String description;
    private boolean publish; // true: public, false: private

    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private LocalDateTime createdAt;

    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private LocalDateTime lastUpdated;

    private UserModel owner;
    private List<PlaylistItem> items;
}

File: views/src/models/PlaylistModel.ts
import { User } from "./User";

export interface PlaylistItem {
  podcastId: string;
  thumbnail: string;
  duration: number;
  title: string;
  description: string;
  owner: string;
  order: number;
}

export interface PlaylistModel {
  id: string;
  name: string;
  description: string;
  thumbnail: string | null;
  publish: boolean;
  lastUpdated: string;
  createdAt: string;
  owner: User;
  items: PlaylistItem[];
  totalItems: number;
  hiddenCount: number;
}

export interface CreatePlaylistDTO {
  name: string;
  description: string;
  publish: boolean;
}